src/OceanSimulations/OceanSimulations.jl

```diff
@@ -96,10 +104,12 @@ function ocean_simulation(grid; Δt = 5minutes,
         momentum_advection = nothing
     end
 
-    tracers = (:T, :S)
+
     if closure isa CATKEVerticalDiffusivity
         tracers = tuple(tracers..., :e)
-        tracer_advection = (; T = tracer_advection, S = tracer_advection, e = nothing)
+        advection_value = tuple(fill(tracer_advection, length(tracers))...)
+        tracer_advection = (; zip(tracers,advection_value)...)
+        tracer_advection = merge(tracer_advection, (e = nothing,))
```

Comment:
Something along these lines might work too:

```julia
tracer_advection = NamedTuple(name => tracer_advection for name in tracers)
tracer_advection = merge(tracer_advection, (; e=nothing))
```

Reply:
Also should we have something that checks if the user has already provided a named tuple? something like

```julia
if !(tracer_advection isa NamedTuple)
    # do this
end
```

That way people can experiment with also advecting TKE which may be useful.

cc @simone-silvestri

Reply:
Dictionaries also can be used which may be a little easier to understand:

```julia
tracer_advection = Dict(name => tracer_advection for name in tracers)
tracer_advection[:e] = nothing
tracer_advection = NamedTuple(name => tracer_advection[name] for name in tracers)
```

Reply:
Thanks @glwagner for the suggestions